### ISSUE TITLE
On ground control

### ISSUE THREAD

**Issue by ArtelGG** [chylex/Better-Sprinting#51]
This is a bit of an odd one to explain, after using this mod for a little while, I can say it's absolutely incredible and an excellent alternative to something like ToggleSneak, however, there's one thing that bugs me immensely regarding the sprint check waiting for the player to be on the ground.

**Keep in mind this post is primarily regarding the 1.7.10 client and implementation of this mod.**

>When a player is hit or hits someone else, their sprint is broken. The mod waits for the player to be on the ground to set the sprint again, this can cause some "issues" with people who use this mod, especially for PvPing. With how fast combat is on 1.7.10, you're going to eventually break sprint when you're airborne because you got hit. Due to how important sprinting is in combat, this can cause you to get wailed on for several hits because you're taking more knockback than your opponent due to you not sprinting towards them.

If you want an example of this, let me know. I can provide a video if needed.

**TL;DR:**  Would it be possible for you to add an option to *always* set the sprint status, regardless if the player is on the ground or not? Even if it's an option hidden within the configuration file, I think it'd be extremely useful.

**Comment by chylex** [comment]
Interesting, that does appear to be a difference compared to vanilla in the old versions. The entire system was rewritten for Better Sprinting 3 which should not have the issue, but is only available for 1.12 and newer.

I don't know if I want to do another round of backporting though, 1.7 is the oldest version I've supported recently and it was a pain to work with. I'll keep the issue open if I decide to do the backport in the future.

**Comment by ArtelGG** [comment]
Think of it this way, if you were to push an update like this once 3.0.0 is considered 100% stable and fully featured, you could do one more round of backporting. Then after that, you could make a decision if you ever want to push an update again, like once every 6 or 12 months or something like that. I've never worked with Forge mods, however, if at some point you ever felt like improving the base for external assistance with pull requests or what not, I'd be happy to try. Regardless, if you do decide not to backport this I'll still try to make a fork and see if I can do it myself :)

**Comment by chylex** [comment]
Backporting to old versions (<1.12) is more of a "once every 5 years" thing. Specifically 1.7 and 1.8 are so old that some of the latest development tools need to be hacked around just to get those versions to work, and every official release needs a lot of testing, so I don't know if I want to bother again :P. Feel free to try the backport, or just fix the bug alone.

**Comment by ArtelGG** [comment]
Either way, you've kept those versions updated for this long where others such as ToggleSneak have gone completely outdated, obfuscated/private forks and other nonsense. You've done it all open source and continue to do so. Much respect, you've done more than enough.
